### ISSUE TITLE
Cannot find Custom Resource Definition error

### ISSUE THREAD

**Issue by ambye85** [java-operator-sdk/java-operator-sdk#274]
I've run the MySQL sample and get the following output:

```shell
java -jar target/mysql-schema-sample-1.6.2-SNAPSHOT.jar
08:50:46.066 INFO  MySQLSchemaOperator.java      (22) MySQL Schema Operator starting
08:50:47.668 INFO  Operator.java                 (112) Registered Controller: 'SchemaController' for CRD: 'class io.javaoperatorsdk.operator.sample.Schema' for namespaces: [all/client namespace]
```

I've copied the sample code into a standalone Gradle project and built a shadow JAR, which gives me the following error (I've got a slightly different log format):

```shell
java -jar libs/demo-operator-0.0.1-SNAPSHOT.jar
Exception in thread "main" java.lang.IllegalArgumentException: No custom resource has been found for controller io.javaoperatorsdk.operator.sample.SchemaController
	at io.javaoperatorsdk.operator.config.runtime.RuntimeControllerMetadata.getCustomResourceClass(RuntimeControllerMetadata.java:32)
	at io.javaoperatorsdk.operator.config.runtime.AnnotationConfiguration.getCustomResourceClass(AnnotationConfiguration.java:49)
	at io.javaoperatorsdk.operator.Operator.registerController(Operator.java:78)
	at io.javaoperatorsdk.operator.Operator.registerControllerForAllNamespaces(Operator.java:56)
	at io.javaoperatorsdk.operator.sample.MySQLSchemaOperator.main(MySQLSchemaOperator.java:27)
```

Has anyone else come across this and what's the solution? Am I just missing something bleedingly obvious from the docs/samples?

**Comment by psycho-ir** [comment]
Hi @ambye85, have you followed the instructions outline [here](https://github.com/java-operator-sdk/java-operator-sdk/tree/master/samples/mysql-schema#build--deploy)?

Most probably at least the crd.yaml hasn't been applied

**Comment by ambye85** [comment]
Hi @psycho-ir. Yes, I have applied the `crd.yaml`. It's exactly the same CRD as `samples/mysql-schema`; all I've done is copied the class definitions into my project so it should be exactly the same. I've tried with `operator-framework:1.6.1` and `operator-framework:1.6.2-SNAPSHOT`; the latter is the version used when I build the samples but it doesn't make any difference in my own project.

**Comment by psycho-ir** [comment]
I just checked the error again, it's about the custom resource class, not its definition. 
The SDK uses an annotation processor to generate the mappings between controllers and custom resource classes. 

I think adding the following dependency to the Gradle would probably solve this issue,  if it didn't would be great if you could put your code in Ggithub and share its link for us to have a look into it. 
```
 annotationProcessor 'io.javaoperatorsdk:operator-framework:1.6.1'
 ```

**Comment by ambye85** [comment]
Ah ha, that solves it, although the generated code causes warnings regarding unchecked exceptions:

```shell
❯ ./gradlew installDist

> Task :compileJava
warning: Supported source version 'RELEASE_8' from annotation processor 'org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor' less than -source '11'
/Users/ashley/code/strataform/strataform-operator/build/generated/sources/annotationProcessor/java/main/io/javaoperatorsdk/operator/sample/SchemaDoneable.java:8: warning: [unchecked] unchecked method invocation: constructor <init> in class CustomResourceDoneable is applied to given types
    super(resource,function);
         ^
  required: T,Function<T,T>
  found:    Schema,Function
  where T is a type-variable:
    T extends CustomResource declared in class CustomResourceDoneable
/Users/ashley/code/strataform/strataform-operator/build/generated/sources/annotationProcessor/java/main/io/javaoperatorsdk/operator/sample/SchemaDoneable.java:8: warning: [unchecked] unchecked conversion
    super(resource,function);
                   ^
  required: Function<T,T>
  found:    Function
  where T is a type-variable:
    T extends CustomResource declared in class CustomResourceDoneable
3 warnings

BUILD SUCCESSFUL in 3s
5 actionable tasks: 4 executed, 1 up-to-date
```

Is this something we can easily fix?

**Comment by psycho-ir** [comment]
> Is this something we can easily fix?

I just opened a [PR](#275) to address these 2 warnings you are getting

**Comment by ambye85** [comment]
Thanks. Appreciate your help. It'd be good to get the docs amended to capture this detail. I'm happy to contribute such a change but am not sure which file would be best to add this too. Any suggestions?

**Comment by psycho-ir** [comment]
> Thanks. Appreciate your help. It'd be good to get the docs amended to capture this detail. I'm happy to contribute such a change but am not sure which file would be best to add this too. Any suggestions?

Thanks for your interesting to contribute to make the SDK better. 
I think an example in the part of the **usage** where it's shown how to add the dependency in maven projects to showcase how it can be done for Gradle would be a good place for it. 

As a side note, we are working on a CLI for the SDK which will take care of initializing an operator project: 
https://github.com/java-operator-sdk/josdk-cli 
It's in early stages but it able to generate sample projects for both Maven & Gradle.

**Comment by psycho-ir** [comment]
The changes are merged and it part of  current `1.6.2-SNAPSHOT`.
